_posts/en/newsletters/2021-03-10-newsletter.md

```diff
@@ -145,7 +145,13 @@ repo], [Hardware Wallet Interface (HWI)][hwi repo],
 [Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
 BOLTs][bolts repo].*
 
-- [Bitcoin Core #20685][] Add I2P support using I2P SAM FIXME:adamjonas
+- [Bitcoin Core #20685][] Adds I2P support by using the
+  [I2P SAM protocol][I2P SAM protocol].  This feature has
+  [long been requested][Bitcoin Core #2091] and was only recently made
+  possible by the addition of [addr v2][addr v2 newsletter mention].
+  Though documentation for node operators hoping to run I2P is still
+  being created, a few [IRC discussions][I2P install comment] provide
```

Comment:
I'm not too confident about adding these IRC discussions but thought it might be nice in the absence of documentation.

Reply:
(if useful, a how-to in one tweet: https://twitter.com/jonatack/status/1366764964896075776, just needs to be i2pd version 2.35 and up)

Reply:
or here https://github.com/bitcoin/bitcoin/pull/20685#pullrequestreview-601031442 though it's not really explicit

Reply:
@jonatack: Cool, added a SE question: https://bitcoin.stackexchange.com/questions/103402/how-can-i-use-bitcoin-core-with-the-anonymous-network-protocol-i2p

Reply:
Thanks! Very handy. One error and a couple changes: `onlynet=tor` should in fact be `onlynet=onion`, and maybe drop the `debug=tor` and `onlynet=onion` (I was replying on twitter to someone asking how to run both together, could keep the tweet link though). Finally, the version of the `i2pd` router apparently needs to be 2.35 and up (2.36 was recently released), at least for now--further I2P code changes are in the works.

Reply:
There are 2 config options for I2P:
```
  -i2pacceptincoming
       If set and -i2psam is also set then incoming I2P connections are
       accepted via the SAM proxy. If this is not set but -i2psam is set
       then only outgoing connections will be made to the I2P network.
       Ignored if -i2psam is not set. Listening for incoming I2P
       connections is done through the SAM proxy, not by binding to a
       local address and port (default: 1)

  -i2psam=<ip:port>
       I2P SAM proxy to reach I2P peers and accept I2P connections (default:
       none)
```

Reply:
I guess I really need to write a doc...

Reply:
@jonatack: Updated, thanks. If you have any further edits, corrections feel free to comment on the SE answer.

Reply:
@michaelfolkson thanks--much appreciated. I don't have an SE account (like LinkedIn, I've never had one) and don't think I can comment without one.